SimCalorimetry/EcalEBTrigPrimAlgos/src/EcalEBTrigPrimClusterAlgo.cc

```diff
@@ -0,0 +1,469 @@
+#include "DataFormats/EcalDigi/interface/EBDataFrame.h"
+#include "DataFormats/EcalDigi/interface/EEDataFrame.h"
+#include "DataFormats/EcalDetId/interface/EcalTrigTowerDetId.h"
+#include "DataFormats/EcalDetId/interface/EBDetId.h"
+#include "DataFormats/EcalDetId/interface/EcalTriggerElectronicsId.h"
+
+#include "CondFormats/EcalObjects/interface/EcalTPGPedestals.h"
+#include "CondFormats/DataRecord/interface/EcalTPGPedestalsRcd.h"
+
+
+
+#include <TTree.h>
+#include <TMath.h>
+\
+
+//----------------------------------------------------------------------
+
+//const unsigned int EcalEBTrigPrimClusterAlgo::nrSamples_=5;
+//const unsigned int EcalEBTrigPrimClusterAlgo::maxNrTowers_=2448;
+//const unsigned int EcalEBTrigPrimClusterAlgo::maxNrSamplesOut_=10;
+
+
+EcalEBTrigPrimClusterAlgo::EcalEBTrigPrimClusterAlgo(const edm::EventSetup & setup,int nSam, int binofmax,bool tcpFormat, bool barrelOnly,bool debug, bool famos)
+
+{
+  barrelOnly_=barrelOnly;
+  binOfMaximum_=binofmax;
+  tcpFormat_=tcpFormat;
+  debug_=debug; 
+  famos_=famos;
+  nSamples_=nSam;
+  maxNrSamples_=10;
+  this->init(setup);
+
+  std::vector <float> etVec;
+  etVec.resize(maxNrSamples_);
+  for (int i=0;i<maxNrSamples_;i++) etVec[i]=0.;
+  clusters_out_.resize(10000);  
+  for (int i=0;i<10000;i++) clusters_out_[i]=etVec;
+
+  fenixTcpFormatClu_ = new EcalFenixTcpFormatCluster(tcpFormat_, debug_, famos_, binOfMaximum_);
+  tcpformat_out_.resize(maxNrSamples_);   
+
+
+}
+
+/*
+//----------------------------------------------------------------------
+void EcalEBTrigPrimClusterAlgo::init(const edm::EventSetup & setup) {
+  if (!barrelOnly_) {
+    //edm::ESHandle<CaloGeometry> theGeometry;
+    //    edm::ESHandle<CaloSubdetectorGeometry> theEndcapGeometry_handle;
+    setup.get<CaloGeometryRecord>().get( theGeometry );
+    setup.get<IdealGeometryRecord>().get(eTTmap_);
+  }
+
+  // initialise data structures
+  initStructures(towerMapEB_);
+  hitTowers_.resize(maxNrTowers_);
+
+
+  linearizer_.resize(nbMaxXtals_);
+  for (int i=0;i<nbMaxXtals_;i++) linearizer_[i] = new  EcalFenixLinearizer(famos_);
+
+  //
+  std::vector <int> v;
+  v.resize(maxNrSamples_);
+  lin_out_.resize(nbMaxXtals_);  
+  for (int i=0;i<5;i++) lin_out_[i]=v;
+  //
+  amplitude_filter_ = new EcalFenixAmplitudeFilter();
+  filt_out_.resize(maxNrSamples_);
+  peak_out_.resize(maxNrSamples_);
+  // these two are dummy
+  fgvb_out_.resize(maxNrSamples_);
+  fgvb_out_temp_.resize(maxNrSamples_);  
+  //
+  peak_finder_ = new  EcalFenixPeakFinder();
+  fenixFormatterEB_ = new EcalFenixStripFormatEB();
+  format_out_.resize(maxNrSamples_);
+  //
+  fenixTcpFormat_ = new EcalFenixTcpFormat(tcpFormat_, debug_, famos_, binOfMaximum_);
+  tcpformat_out_.resize(maxNrSamples_);   
+
+}
+//----------------------------------------------------------------------
+*/
+
+
+
+EcalEBTrigPrimClusterAlgo::~EcalEBTrigPrimClusterAlgo() 
+{
+  for (int i=0;i<nbMaxXtals_;i++) delete linearizer_[i]; 
+  delete amplitude_filter_;
```

Comment:
it would be better to make these members into smart pointers to avoid possible memory leaks